src/tools/prun/prte-map.1.md

```diff
@@ -0,0 +1,498 @@
+# NAME
+
+PRRTE - Mapping / Binding / Ranking Processes.
+
+# SYNOPSIS
+
+PRRTE employs a three-phase procedure for assigning process locations and ranks:
+
+  1. **mapping**:   Assigns a default location to each process
+  2. **ranking**:   Assigns a unique rank value to each process
+  3. **binding**:   Constrains each process to run on specific processors
+
+This document describes these three phases with examples.
+
+
+# QUICK SUMMARY
+
+The two binaries that most influence process layout are `prte` and `prun`.
+The `prte` process discovers the allocation, starts the daemons, and defines the
+default mapping/ranking/binding for all jobs.
+The `prun` process defines the specific mapping/ranking/binding for a specific
+job. Most of the command line controls are targeted to `prun` since each job has
+its own unique requirements.
+
+The `prte` process attempts to automatically discover the nodes in the
+allocation by querying supported resource managers. If a support resource
+manager is not present then `prte` relies on a hostfile provided by the user.
+In the absence of such a hostfile it will run all processes on the localhost.
+
+If running under a supported resource manager, the `prte` process will start the
+daemon processes (`prted`) on the remote nodes using the corresponding resource
+manager process starter. If no such starter is available then `rsh` or `ssh`
+is used.
+
+Nodes are scheduled (by default) in a round-robin fashion by CPU slot.
```

Comment:
Clarify that this is the default `--map-by` behavior

Reply:
It seems that we might follow the `--bind-to` rule even with `--map-by`

```
[jjhursey@c712f6n01 ~] prterun --np 2  --bind-to package:REPORT hostname
[c712f6n01:47498] MCW rank 0 bound to package[0][core:0-9]
[c712f6n01:47498] MCW rank 1 bound to package[0][core:0-9]
c712f6n01
c712f6n01
[jjhursey@c712f6n01 ~] prterun --np 3  --bind-to package:REPORT hostname
[c712f6n01:47486] MCW rank 0 bound to package[0][core:0-9]
[c712f6n01:47486] MCW rank 1 bound to package[1][core:10-19]
[c712f6n01:47486] MCW rank 2 bound to package[0][core:0-9]
c712f6n01
c712f6n01
c712f6n01
```

```
[jjhursey@c712f6n01 ~] prterun --np 2  --bind-to core:REPORT hostname
[c712f6n01:47579] MCW rank 0 bound to package[0][core:0]
[c712f6n01:47579] MCW rank 1 bound to package[0][core:1]
c712f6n01
c712f6n01
[jjhursey@c712f6n01 ~] prterun --np 3  --bind-to core:REPORT hostname
[c712f6n01:47588] MCW rank 0 bound to package[0][core:0]
[c712f6n01:47588] MCW rank 1 bound to package[1][core:10]
[c712f6n01:47588] MCW rank 2 bound to package[0][core:1]
c712f6n01
c712f6n01
c712f6n01
```

Reply:
I'm not entirely sure what you are asking. The default mapping policy is map-by core when np <= 2, and map-by package when np > 2. The binding policy is orthogonal to it, but has its own default of bind-to core when np <= 2, and bind-to package when np > 2.

Which is what you show above, so it looks like all is good.

Reply:
Yeah. I'm just saying that we document the --bind-to defaults, but not the --map-by defaults. So I'll need to add that to the document.